client-sdk/go/types/token.go

```diff
@@ -87,18 +87,38 @@ func NewBaseUnits(amount quantity.Quantity, denomination Denomination) BaseUnits
 	}
 }
 
+// FormatNamedAddress returns a human-friendly representation of an address:
+// it prints the name (if known) followed by the preferred form of the address
+// in parentheses. If an Ethereum hex address mapping is provided for the native
+// address, it is used; otherwise the native Bech32 address is used.
+func FormatNamedAddress(ctx context.Context, addr Address) string {
```

Comment:
I think that we cannot get rid of the `ctx` here, but I will still comment that fetching things from a `ctx` should be avoided as much as possible, for multiple reasons:
- it is harder to unit test this function,
- the function signature is misleading as things are hidden in the context.

Ideally, this function should actually accept two maps and an address. Or even better, a struct should accept maps, and a method should accept an address. 

Can we add a unit test for this?